openedx/core/djangoapps/discussions/views.py

```diff
@@ -1,22 +1,31 @@
 """
 Handle view-logic for the discussions app.
 """
+from typing import Dict
+
+import edx_api_doc_tools as apidocs
 from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
 from edx_rest_framework_extensions.auth.session.authentication import SessionAuthenticationAllowInactiveUser
+from rest_framework.exceptions import ValidationError
+from rest_framework.request import Request
 from rest_framework.response import Response
 from rest_framework.views import APIView
 
+from lms.djangoapps.discussion.toggles import ENABLE_DISCUSSIONS_MFE
 from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
 from openedx.core.lib.api.authentication import BearerAuthenticationAllowInactiveUser
 from openedx.core.lib.api.view_utils import validate_course_key
-from .models import DiscussionsConfiguration
-from .permissions import check_course_permissions, IsStaffOrCourseTeam
-from .serializers import DiscussionsConfigurationSerializer
+from .models import AVAILABLE_PROVIDER_MAP, DiscussionsConfiguration, Features, Provider
+from .permissions import IsStaffOrCourseTeam, check_course_permissions
+from .serializers import (
+    DiscussionsConfigurationSerializer,
+    DiscussionsProvidersSerializer,
+)
 
 
 class DiscussionsConfigurationView(APIView):
```

Comment:
Don't you think the name of the view should match with the URL? e.g. `DiscussionsConfigurationSettingsView` to make it more readable?